src/backend/metal/src/command.rs

```diff
@@ -3015,28 +3137,27 @@ impl com::RawCommandBuffer<Backend> for CommandBuffer {
         offset: u32,
         constants: &[u32],
     ) {
-        self.update_push_constants(offset, constants);
+        self.state.update_push_constants(offset, constants);
+        let id = self.shared.push_constants_buffer_id;
 
         if stages.intersects(pso::ShaderStageFlags::GRAPHICS) {
             // Note: it's a waste to heap allocate the bytes here in case
             // of no active render pass.
             // Note: the whole range is re-uploaded, which may be inefficient
-            let com_vs = if stages.contains(pso::ShaderStageFlags::VERTEX) {
-                Some(self.push_vs_constants())
-            } else {
-                None
-            };
-            let com_ps = if stages.contains(pso::ShaderStageFlags::FRAGMENT) {
-                Some(self.push_ps_constants())
-            } else {
-                None
-            };
-            let commands = com_vs.into_iter().chain(com_ps);
-
-            self.inner
-                .borrow_mut()
-                .sink()
-                .pre_render_commands(commands);
+            if stages.contains(pso::ShaderStageFlags::VERTEX) {
```

Comment:
The old approach avoids two calls to `pre_render_commands` here – not sure how the `Option` compares to this trade-off

Reply:
I had to go through quite a few hoops with the borrow checker...